python-service/src/circuit_breaker.py

```
@@ -0,0 +1,304 @@
+            remaining = self.config.timeout_seconds - (time.time() - (self._opened_at or 0))
+            raise CircuitBreakerOpenError(self.name, max(0, remaining))
+
+        start_time = time.time()
+        self.metrics.total_calls += 1
+
+        try:
+            result = operation()
+            duration = time.time() - start_time
+            self._record_success(duration)
+            return result
+        except Exception as e:
```

Flake8 (7.3.0)

[error] local variable 'e' is assigned to but never used

(F841)